src/orquestra/sdk/_base/_in_process_runtime.py

```diff
@@ -95,6 +96,10 @@ def create_workflow_run(self, workflow_def: ir.WorkflowDef) -> WfRunId:
             id: deserialize_constant(node)
             for id, node in workflow_def.constant_nodes.items()
         }
+        for id, secret in workflow_def.secret_nodes.items():
+            consts[id] = secrets.get(
```

Comment:
**thought**: I like the conciseness of dumping secrets as constants instead of a separate collection, but let's keep in mind this is prone to ID collision bugs if we ever generate `ConstantNodeID` and `SecretNodeID` from similar value domains.

Reply:
Yeah, if we ever have ID collisions then this would cause issues. ID collisions would cause problems in other places too, I think.